spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/AbstractJdbcConfiguration.java

```diff
@@ -106,26 +120,45 @@ public JdbcCustomConversions jdbcCustomConversions() {
 	 */
 	@Bean
 	public JdbcAggregateTemplate jdbcAggregateTemplate(ApplicationEventPublisher publisher,
-			RelationalMappingContext context, JdbcConverter converter, DataAccessStrategy dataAccessStrategy) {
+			RelationalMappingContext context, JdbcConverter converter,
+			ObjectProvider<DataAccessStrategy> dataAccessStrategyProvider, NamedParameterJdbcOperations operations,
+			Optional<NamingStrategy> namingStrategy, @Lazy JdbcConverter jdbcConverter) {
+
+		DataAccessStrategy dataAccessStrategy = dataAccessStrategyProvider.getIfAvailable(() -> dataAccessStrategy( //
+				operations, //
+				jdbcConverter, //
+				jdbcMappingContext(namingStrategy)) //
+		);
 
-		return new JdbcAggregateTemplate(publisher, context, converter, dataAccessStrategy);
+		return new JdbcAggregateTemplate( //
```

Comment:
Same here. I think the single line reads way better than the ones broken out onto multiple ones.